### PR TITLE
Refactored Log.cs to a separate project

### LINKED ISSUE #394: Logging as a seperate subproject
Logging in the `ExampleBot` subproject is done through it's `Log` class. 

`SteamTrade` can get their events printed in the trade manager console by throwing an `OnError(string message)` event that a `UserHandler` catches and runs `OnTradeError([...])` on. 

There is no direct way to log events outside of `ExampleBot`, because that is the namespace where `Log.cs` resides.

I have a working local branch where I put `Log.cs` into a separate subproject called `SteamBotLog` so that you can log from everywhere, as long as an instance of the `Log` class is in scope. The drawback of this is that you have to change all your constructors to let you pass in a `Log`.

Is this worth committing?

### PR DESCRIPTION
**Pull request by AleksaSavic3**
I refactored Log.cs to a separate project called Logger.

I fixed all references and namespaces and I also replaced Console.WriteLine() in SteamTrade with Log functions accordingly.

Fixes #394.

### CODE REVIEW

File: SteamTrade/SteamWeb.cs [hunk 1]
```diff
@@ -122,7 +125,7 @@ public bool DoLogin(string username, string password)
             string steamGuardId = "";
             do
             {
-                Console.WriteLine("SteamWeb: Logging In...");
+                Log.Info("SteamWeb: Logging In...");
```
Comment (BlueRaja) [inline SteamTrade/SteamWeb.cs]: This will cause a `NullReferenceException`, no?  `Log` has never been assigned.

File: SteamTrade/Trade.cs [hunk 2]
```diff
@@ -551,9 +554,8 @@ private T RetryWebRequest<T>(Func<T> webEvent)
                 }
                 catch(Exception ex)
                 {
-                    // TODO: log to SteamBot.Log but... see issue #394
                     // realistically we should not throw anymore
-                    Console.WriteLine(ex);
+                    Log.Error(ex.ToString());
```
Comment (BlueRaja) [inline SteamTrade/Trade.cs]: Same as above

Comment (AleksaSavic3) [inline SteamTrade/Trade.cs]: Oh, I see. I will fix it.

File: SteamTrade/TradeManager.cs [hunk 3]
```diff
@@ -361,11 +363,10 @@ private bool CheckTradeTimeout (Trade trade)
         [Conditional ("DEBUG_TRADE_MANAGER")]
         private static void DebugPrint (string output)
         {
-            // I don't really want to add the Logger as a dependecy to TradeManager so I 
-            // print using the console directly. To enable this for debugging put this:
+            // To enable this for debugging put this:
             // #define DEBUG_TRADE_MANAGER
             // at the first line of this file.
-            System.Console.WriteLine (output);
+            Log.Info(output);
```
Comment (WildCard65) [inline SteamTrade/TradeManager.cs]: I think Debug is more suited then Info here

Comment (AleksaSavic3) [inline SteamTrade/TradeManager.cs]: Fixed. I will push more commits soon.

**Comment by AleksaSavic3** [comment]
This PR is too much complicated, should I rebase it?

**Comment by BlueRaja** [comment]
Rather than creating _another_ project to hold the log configuration and creating a new Log instance in every class, why not just use a single static Log instance?

**Comment by AleksaSavic3** [comment]
That would be better, but due to some real life changes I can no longer work on this PR. If someone wants to resume where I left off, feel free.  :relieved:

**Pull request closed without merging**